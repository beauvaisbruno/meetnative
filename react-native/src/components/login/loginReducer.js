"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
const reducerHelper_1 = require("../../utils/reducerHelper");
var LoginActions;
(function(LoginActions) {
  LoginActions["loginGoogle"] = "loginGoogle";
  LoginActions["loginFacebook"] = "loginFacebook";
  LoginActions["loginWithEmail"] = "loginWithEmail";
  LoginActions["loginLoadUser"] = "loginLoadUser";
  LoginActions["loginResetEmail"] = "loginResetEmail";
})((LoginActions = exports.LoginActions || (exports.LoginActions = {})));
const initialState = {
  loginGoogle: new reducerHelper_1.ActionResult(),
  loginFacebook: new reducerHelper_1.ActionResult(),
  loginWithEmail: new reducerHelper_1.ActionResult(),
  loginLoadUser: new reducerHelper_1.ActionResult(),
  loginResetEmail: new reducerHelper_1.ActionResult()
};
function login(state = initialState, action) {
  state = reducerHelper_1.reduceAction(LoginActions, action, state);
  return state;
}
exports.login = login;
