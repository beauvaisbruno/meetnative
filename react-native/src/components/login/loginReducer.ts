import {
  ActionResult,
  IActionResult,
  reduceAction
} from "../../utils/reducerHelper";

export enum LoginActions {
  loginGoogle = "loginGoogle",
  loginFacebook = "loginFacebook",
  loginWithEmail = "loginWithEmail",
  loginLoadUser = "loginLoadUser",
  loginResetEmail = "loginResetEmail"
}
export interface ILoginState {
  readonly loginGoogle: IActionResult<any>;
  readonly loginFacebook: IActionResult<any>;
  readonly loginWithEmail: IActionResult<any>;
  readonly loginLoadUser: IActionResult<any>;
  readonly loginResetEmail: IActionResult<undefined>;
}

const initialState: ILoginState = {
  loginGoogle: new ActionResult<any>(),
  loginFacebook: new ActionResult<any>(),
  loginWithEmail: new ActionResult<any>(),
  loginLoadUser: new ActionResult<any>(),
  loginResetEmail: new ActionResult<undefined>()
};

export function login(state = initialState, action: any) {
  state = reduceAction(LoginActions, action, state);
  return state;
}
