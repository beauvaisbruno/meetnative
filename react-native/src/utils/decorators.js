"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
const redux_1 = require("redux");
const rootReducers_1 = require("../rootReducers");
const redux_devtools_extension_1 = require("redux-devtools-extension");
const react_1 = __importDefault(require("react"));
const react_navigation_1 = require("react-navigation");
const react_redux_1 = require("react-redux");
const native_base_1 = require("native-base");
const mocks_1 = require("./mocks");
const test_utils_1 = require("react-apollo/test-utils");
exports.mockStore = redux_1.createStore(rootReducers_1.rootReducers, redux_devtools_extension_1.composeWithDevTools(redux_1.applyMiddleware()));
exports.mockBase = (children) => class MockBase extends react_1.default.Component {
    render() {
        return react_1.default.createElement(native_base_1.Root, null, children);
    }
};
exports.MockBaseAndNavigator = ({ children }) => {
    const Container = react_navigation_1.createAppContainer(react_navigation_1.createSwitchNavigator({ screen: exports.mockBase(children) }, { initialRouteName: "screen" }));
    return react_1.default.createElement(Container, null);
};
exports.storeBaseAndNavigatorDecorator = (story) => (react_1.default.createElement(react_redux_1.Provider, { store: exports.mockStore },
    react_1.default.createElement(exports.MockBaseAndNavigator, null, story())));
exports.baseDecorator = (story) => react_1.default.createElement(native_base_1.Root, null, story());
exports.navigatorDecorator = (story) => (react_1.default.createElement(exports.MockBaseAndNavigator, null, story()));
exports.storeDecorator = (story) => (react_1.default.createElement(react_redux_1.Provider, { store: exports.mockStore }, story()));
exports.MockedApollo = ({ children, mocks }) => {
    const formatedMocks = [];
    mocks.forEach(mock => {
        const formatedMock = {
            request: {
                query: mock.q,
                variables: mock.v
            },
            result: {
                data: mock.r,
                errors: mock.e ? [{ message: "Mocked Error" }] : undefined
            }
        };
        formatedMocks.push(formatedMock);
    });
    return (react_1.default.createElement(test_utils_1.MockedProvider, Object.assign({}, {
        mocks: formatedMocks,
        cache: mocks_1.createCache()
    }), children));
};
