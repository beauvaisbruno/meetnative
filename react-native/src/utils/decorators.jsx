"use strict";
var __extends = (this && this.__extends) || (function () {
    var extendStatics = function (d, b) {
        extendStatics = Object.setPrototypeOf ||
            ({ __proto__: [] } instanceof Array && function (d, b) { d.__proto__ = b; }) ||
            function (d, b) { for (var p in b) if (b.hasOwnProperty(p)) d[p] = b[p]; };
        return extendStatics(d, b);
    };
    return function (d, b) {
        extendStatics(d, b);
        function __() { this.constructor = d; }
        d.prototype = b === null ? Object.create(b) : (__.prototype = b.prototype, new __());
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
var redux_1 = require("redux");
var rootReducers_1 = require("../rootReducers");
var redux_devtools_extension_1 = require("redux-devtools-extension");
var react_1 = require("react");
var react_navigation_1 = require("react-navigation");
var react_redux_1 = require("react-redux");
var native_base_1 = require("native-base");
var mocks_1 = require("./mocks");
var test_utils_1 = require("react-apollo/test-utils");
exports.mockStore = redux_1.createStore(rootReducers_1.rootReducers, redux_devtools_extension_1.composeWithDevTools(redux_1.applyMiddleware()));
exports.mockBase = function (children) {
    return /** @class */ (function (_super) {
        __extends(MockBase, _super);
        function MockBase() {
            return _super !== null && _super.apply(this, arguments) || this;
        }
        MockBase.prototype.render = function () {
            return <native_base_1.Root>{children}</native_base_1.Root>;
        };
        return MockBase;
    }(react_1.default.Component));
};
exports.MockBaseAndNavigator = function (_a) {
    var children = _a.children;
    var Container = react_navigation_1.createAppContainer(react_navigation_1.createSwitchNavigator({ screen: exports.mockBase(children) }, { initialRouteName: "screen" }));
    return <Container />;
};
exports.storeBaseAndNavigatorDecorator = function (story) { return (<react_redux_1.Provider store={exports.mockStore}>
    <exports.MockBaseAndNavigator>{story()}</exports.MockBaseAndNavigator>
  </react_redux_1.Provider>); };
exports.baseDecorator = function (story) { return <native_base_1.Root>{story()}</native_base_1.Root>; };
exports.navigatorDecorator = function (story) { return (<exports.MockBaseAndNavigator>{story()}</exports.MockBaseAndNavigator>); };
exports.storeDecorator = function (story) { return (<react_redux_1.Provider store={exports.mockStore}>{story()}</react_redux_1.Provider>); };
exports.MockedApollo = function (_a) {
    var children = _a.children, mocks = _a.mocks;
    var formatedMocks = [];
    mocks.forEach(function (mock) {
        // erros: [{ message: "Error!" }]
        var formatedMock = {
            request: {
                query: mock.q,
                variables: mock.v
            },
            result: {
                data: mock.r,
                errors: mock.e ? [{ message: "Mocked Error" }] : undefined
            }
        };
        formatedMocks.push(formatedMock);
    });
    return (<test_utils_1.MockedProvider {...{
        mocks: formatedMocks,
        cache: mocks_1.createCache()
    }}>
      {children}
    </test_utils_1.MockedProvider>);
};
