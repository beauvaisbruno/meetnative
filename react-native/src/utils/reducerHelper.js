"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
const typesafe_actions_1 = require("typesafe-actions");
class ActionResult {
  constructor(defaultData) {
    this.fetching = false;
    this.error = false;
    if (defaultData !== undefined) this.data = defaultData;
  }
}
exports.ActionResult = ActionResult;
exports.successAction = (type, values, meta) => {
  const act = typesafe_actions_1.action(type + "Success", values, meta);
  return act;
};
exports.errorAction = (type, values, meta) => {
  const act = typesafe_actions_1.action(type + "Error", values, meta);
  return act;
};
function getNextState(action) {
  let state = { fetching: true, error: false };
  if (action.type.includes("Success")) {
    state = { fetching: false, error: false };
    if (action.payload !== undefined)
      state = Object.assign({}, state, { data: action.payload });
  }
  if (action.type.includes("Error"))
    state = {
      fetching: false,
      error: true,
      errorData: action.payload ? action.payload : undefined
    };
  return state;
}
exports.getNextState = getNextState;
function reduceAction(actions, action, state) {
  const fieldName = action.type.replace("Error", "").replace("Success", "");
  if (!actions[fieldName]) return state;
  const prevState = state[fieldName]
    ? Object.assign({}, state[fieldName])
    : { fetching: false, error: false };
  const parentState = Object.assign({}, state);
  parentState[fieldName] = Object.assign({}, prevState, getNextState(action));
  return parentState;
}
exports.reduceAction = reduceAction;
exports.fetchingAction = (type, values, meta) => {
  return typesafe_actions_1.action(type, values, meta);
};
