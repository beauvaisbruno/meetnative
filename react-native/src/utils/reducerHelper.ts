import { action as actionCreator } from "typesafe-actions";
import { PayloadMetaAction } from "typesafe-actions/dist/type-helpers";

export type IAction = PayloadMetaAction<string, any, any>;
export type IActionData<T> = PayloadMetaAction<string, T, any>;

export interface IActionResult<T> {
  fetching: boolean;
  error: boolean;
  data?: T;
  errorData?: any;
}

export class ActionResult<T> implements IActionResult<T> {
  error: boolean;
  fetching: boolean;
  data?: T;
  constructor(defaultData?: T) {
    this.fetching = false;
    this.error = false;
    if (defaultData !== undefined) this.data = defaultData;
  }
}

export const successAction = (type: string, values?: any, meta?: any): any => {
  const act: any = actionCreator(type + "Success", values, meta);
  return act;
};
export const errorAction = (type: string, values?: any, meta?: any): any => {
  const act: any = actionCreator(type + "Error", values, meta);
  return act;
};

export function getNextState(action: IAction): IActionResult<any> {
  let state: IActionResult<any> = { fetching: true, error: false };
  if (action.type.includes("Success")) {
    state = { fetching: false, error: false };
    if (action.payload !== undefined)
      state = { ...state, data: action.payload };
  }
  if (action.type.includes("Error"))
    state = {
      fetching: false,
      error: true,
      errorData: action.payload ? action.payload : undefined
    };
  return state;
}
export function reduceAction<T extends { [key: string]: any }>(
  actions: any,
  action: IAction,
  state: T
): T {
  const fieldName = action.type.replace("Error", "").replace("Success", "");
  if (!actions[fieldName]) return state;
  const prevState: IActionResult<any> = state[fieldName]
    ? { ...state[fieldName] }
    : { fetching: false, error: false };
  const parentState = { ...state };

  parentState[fieldName] = { ...prevState, ...getNextState(action) };
  return parentState;
}

export const fetchingAction = (type: string, values?: any, meta?: any): any => {
  return actionCreator(type, values, meta);
};
