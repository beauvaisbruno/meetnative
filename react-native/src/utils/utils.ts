import { TeachLevel } from "../TeachLevel";
import { LearnLevel } from "../LearnLevel";
import geolib from "geolib";

export const MIN = 60 * 1000;
export const HOUR = 60 * MIN;
export const DAY = 24 * HOUR;

export const sortByDistance = (
  userLat: number,
  userLon: number,
  profilesData: { lat: number; lon: number }[]
) => {
  return profilesData.sort((a, b) => {
    const aDistance = distanceKm(userLat, userLon, a.lat, a.lon);
    const bDistance = distanceKm(userLat, userLon, b.lat, b.lon);
    if (aDistance < bDistance) return -1;
    if (aDistance > bDistance) return 1;
    return 0;
  });
};
export const sortByTimeAsc = (data: { time: number }[]) => {
  return data.sort((a: { time: number }, b: { time: number }) => {
    if (a.time < b.time) return -1;
    if (a.time > b.time) return 1;
    return 0;
  });
};

export function objsToArrayWithKey(obj: { [x: string]: any }) {
  return Object.keys(obj).map(key => {
    obj[key].key = key;
    return obj[key];
  });
}

export function isEmpty(value: any): boolean {
  const isEmptyObject = (a: any): boolean => {
    if (typeof a.length === "undefined") {
      // it's an Object, not an Array
      const hasNonempty = Object.keys(a).some(element => !isEmpty(a[element]));
      return hasNonempty ? false : isEmptyObject(Object.keys(a));
    }

    return !a.some((element: any) => {
      // check if array is really not empty as JS thinks
      return !isEmpty(element); // at least one element should be non-empty
    });
  };
  return (
    !value ||
    typeof value === "undefined" ||
    value === null ||
    (typeof value === "object" && isEmptyObject(value))
  );
}

export function getRandomProp<T>(obj: { [key: string]: T }): T {
  const keys = Object.keys(obj);
  const randKey = keys[Math.floor(Math.random() * keys.length)];
  return obj[randKey];
}
export function getRandomKey(obj: { [key: string]: any }): string {
  const keys = Object.keys(obj);
  const key = keys[Math.floor(Math.random() * keys.length)];
  return key;
}
export function intersectObjKey(
  obj1: { [key: string]: any },
  obj2: { [key: string]: any }
): string[] {
  return Object.keys(obj1).filter(function(n) {
    return Object.keys(obj2).indexOf(n) > -1;
  });
}

export function langMatch(
  learnLangsLevels: { [p: string]: string },
  teachLangsLevels: { [p: string]: string }
): boolean {
  const learnLangs: string[] = Object.keys(learnLangsLevels);
  const teachLangs: string[] = Object.keys(teachLangsLevels);
  for (const learnLang of learnLangs) {
    if (
      teachLangs.indexOf(learnLang) !== -1 &&
      (teachLangsLevels[learnLang] !== TeachLevel.fluent ||
        learnLangsLevels[learnLang] !== LearnLevel.fluent)
    )
      return true;
  }
  return false;
}
export function distanceKm(
  fromLat: number,
  fromLon: number,
  toLat: number,
  toLon: number
): number {
  return Math.round(
    geolib.getDistance(
      { latitude: fromLat, longitude: fromLon },
      { latitude: toLat, longitude: toLon }
    ) / 1000
  );
}
