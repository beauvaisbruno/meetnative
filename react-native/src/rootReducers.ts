import { AnyAction, combineReducers } from "redux";
import { IProfileState, profile } from "./components/profile/profileReducer";
import { ILoginState, login } from "./components/login/loginReducer";
// @ts-ignore
import { alertReducer } from "redux-saga-rn-alert";
import { ISearchState, search } from "./components/search/searchReducer";
import { IRoomState, room } from "./components/room/roomReducer";

export interface IApplicationState {
  profile: IProfileState;
  login: ILoginState;
  search: ISearchState;
  room: IRoomState;
}

const appReducer = combineReducers<IApplicationState & any>({
  profile,
  login,
  search,
  room,
  alertReducer
});

export const ClearAll = "ClearAll";

export const rootReducers = (
  state: IApplicationState | undefined,
  action: AnyAction
) => {
  if (action.type === ClearAll) {
    state = undefined;
  }
  return appReducer(state, action);
};

export interface IProfile {
  langsToLearn: { [key: string]: string };
  langsToTeach: { [key: string]: string };
  age: number;
  pseudo: string;
  profession: string;
  city: string;
  avatar: string;
  isMale: boolean;
  lat?: number;
  lon?: number;
  geohash?: number;
  id?: string;
  fake?: boolean;
  fcmId?: string;
}
