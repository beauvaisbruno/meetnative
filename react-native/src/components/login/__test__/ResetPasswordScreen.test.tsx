import { ValidationError } from "yup";
import ResetPasswordScreenContainer, {
  ResetPasswordScreen,
  FormUi,
  getValidationSchema,
  IUser,
  mapStateToProps,
  mapDispatchToProps
} from "../ResetPasswordScreen";
import { shallow } from "enzyme";
import * as React from "react";
import { NavigationInjectedProps } from "react-navigation";
import { FormikActions, InjectedFormikProps } from "formik";
import toJson from "enzyme-to-json";
import { IApplicationState } from "../../../rootReducers";
import * as reducerHelper from "../../../utils/reducerHelper";
import { LoginActions } from "../loginReducer";

it("ValidationSchema", () => {
  let fields = { email: "email@provider.com" };
  expect(getValidationSchema().validateSync(fields)).toBe(fields);
  fields = { email: "email@provider" };
  expect(() => getValidationSchema().validateSync(fields)).toThrowError(
    new ValidationError("E-mail is not valid!", fields, "email")
  );
});

it("FormUi", () => {
  const handleSubmit = jest.fn();
  const setFieldTouched = jest.fn(() => null);
  const handleChangeImpl = jest.fn(() => null);
  const tree = shallow(
    // @ts-ignore
    <FormUi
      fetching={false}
      error={false}
      values={{ email: "" }}
      errors={{}}
      touched={{}}
      handleChange={jest.fn(() => handleChangeImpl)}
      setFieldTouched={setFieldTouched}
      handleSubmit={handleSubmit}
    />
  );
  expect(toJson(tree)).toMatchSnapshot();
  const submitBtn = tree.findWhere(node => node.prop("testID") === "resetBtn");
  submitBtn.props().onPress();
  expect(handleSubmit.mock.calls.length).toBe(1);

  tree.setProps({ fetching: true });
  expect(toJson(tree)).toMatchSnapshot();
  tree.setProps({
    touched: { pseudo: true },
    errors: { pseudo: "error message" }
  });
  expect(toJson(tree)).toMatchSnapshot();

  const inputs: any = tree.find("Styled(Input)");
  inputs.props().onBlur();
  inputs.props().onChangeText();
  expect(setFieldTouched.mock.calls.length).toBe(1);
  expect(handleChangeImpl.mock.calls.length).toBe(1);
});

it("Formik wrapper", () => {
  const resetEmail = jest.fn();
  const tree = shallow(
    //@ts-ignore
    <ResetPasswordScreen
      errors={{}}
      error={false}
      fetching={false}
      resetEmail={resetEmail}
    />
  );
  expect(toJson(tree)).toMatchSnapshot();
  // @ts-ignore
  tree
    .find("Formik")
    .props()
    .onSubmit();
  expect(resetEmail).toBeCalled();
});

it("Container", () => {
  // @ts-ignore
  const wrapper = shallow(<ResetPasswordScreen />);
  expect(wrapper).toBeDefined();
  expect(
    mapStateToProps(({
      login: { loginResetEmail: { expected: "expected" } }
    } as unknown) as IApplicationState)
  ).toEqual({ expected: "expected" });

  const dispatchMock = jest.fn();
  const fetchingAction = jest.fn().mockReturnValue("fetchingAction");
  jest
    .spyOn(reducerHelper, "fetchingAction")
    .mockImplementation(fetchingAction);

  const map = mapDispatchToProps(dispatchMock);
  map.resetEmail(
    ("values" as unknown) as IUser,
    ("actions" as unknown) as FormikActions<IUser>
  );
  expect(dispatchMock).lastCalledWith("fetchingAction");
  expect(fetchingAction).lastCalledWith(
    LoginActions.loginResetEmail,
    "values",
    "actions"
  );
});
