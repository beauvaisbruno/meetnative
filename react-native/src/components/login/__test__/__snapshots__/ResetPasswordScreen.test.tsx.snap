// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FormUi 1`] = `
<Styled(ViewNB)>
  <Styled(ViewNB)
    padder={true}
  >
    <Styled(Item)
      stackedLabel={true}
    >
      <Styled(Label)>
        Email
      </Styled(Label)>
      <Styled(Input)
        onBlur={[Function]}
        onChangeText={[MockFunction]}
        value=""
      />
    </Styled(Item)>
  </Styled(ViewNB)>
  <Styled(ViewNB)
    padder={true}
  >
    <Styled(Button)
      block={true}
      onPress={[Function]}
      rounded={true}
      success={true}
      testID="resetBtn"
    >
      <Styled(Icon)
        color="white"
        fontSize={30}
        name="lock-reset"
        type="MaterialCommunityIcons"
      />
      <Styled(Text)>
        Reset the password
      </Styled(Text)>
    </Styled(Button)>
  </Styled(ViewNB)>
</Styled(ViewNB)>
`;

exports[`FormUi 2`] = `
<Styled(ViewNB)>
  <Styled(ViewNB)
    padder={true}
  >
    <Styled(Item)
      stackedLabel={true}
    >
      <Styled(Label)>
        Email
      </Styled(Label)>
      <Styled(Input)
        onBlur={[Function]}
        onChangeText={[MockFunction]}
        value=""
      />
    </Styled(Item)>
  </Styled(ViewNB)>
  <Styled(ViewNB)
    padder={true}
  >
    <Styled(Button)
      block={true}
      onPress={[Function]}
      rounded={true}
      success={true}
      testID="resetBtn"
    >
      <Styled(Icon)
        color="white"
        fontSize={30}
        name="spinner"
        type="EvilIcons"
      />
      <Styled(Text)>
        Reset the password
      </Styled(Text)>
    </Styled(Button)>
  </Styled(ViewNB)>
</Styled(ViewNB)>
`;

exports[`FormUi 3`] = `
<Styled(ViewNB)>
  <Styled(ViewNB)
    padder={true}
  >
    <Styled(Item)
      stackedLabel={true}
    >
      <Styled(Label)>
        Email
      </Styled(Label)>
      <Styled(Input)
        onBlur={[Function]}
        onChangeText={[MockFunction]}
        value=""
      />
    </Styled(Item)>
  </Styled(ViewNB)>
  <Styled(ViewNB)
    padder={true}
  >
    <Styled(Button)
      block={true}
      onPress={[Function]}
      rounded={true}
      success={true}
      testID="resetBtn"
    >
      <Styled(Icon)
        color="white"
        fontSize={30}
        name="spinner"
        type="EvilIcons"
      />
      <Styled(Text)>
        Reset the password
      </Styled(Text)>
    </Styled(Button)>
  </Styled(ViewNB)>
</Styled(ViewNB)>
`;

exports[`Formik wrapper 1`] = `
<Styled(ViewNB)>
  <ErrorableImage
    source={
      Object {
        "testUri": "../../../src/components/login/img/header.png",
      }
    }
    width={750}
  />
  <Formik
    enableReinitialize={false}
    initialValues={
      Object {
        "connection": null,
        "email": "",
      }
    }
    isInitialValid={false}
    onSubmit={[Function]}
    validateOnBlur={true}
    validateOnChange={true}
    validationSchema={
      ObjectSchema {
        "_blacklist": RefSet {
          "list": Set {},
          "refs": Map {},
        },
        "_conditions": Array [],
        "_defaultDefault": [Function],
        "_deps": Array [],
        "_excludedEdges": Array [],
        "_exclusive": Object {},
        "_mutate": false,
        "_nodes": Array [
          "email",
        ],
        "_options": Object {
          "abortEarly": true,
          "recursive": true,
        },
        "_type": "object",
        "_typeError": [Function],
        "_whitelist": RefSet {
          "list": Set {},
          "refs": Map {},
        },
        "fields": Object {
          "email": StringSchema {
            "_blacklist": RefSet {
              "list": Set {},
              "refs": Map {},
            },
            "_conditions": Array [],
            "_deps": Array [],
            "_exclusive": Object {
              "required": false,
              "undefined": false,
            },
            "_mutate": false,
            "_options": Object {
              "abortEarly": true,
              "recursive": true,
            },
            "_type": "string",
            "_typeError": [Function],
            "_whitelist": RefSet {
              "list": Set {},
              "refs": Map {},
            },
            "tests": Array [
              [Function],
              [Function],
              [Function],
            ],
            "transforms": Array [
              [Function],
            ],
          },
        },
        "tests": Array [],
        "transforms": Array [
          [Function],
        ],
      }
    }
  >
    <Component />
  </Formik>
</Styled(ViewNB)>
`;
